go/vt/vttablet/tabletserver/messager/engine.go

```diff
@@ -106,6 +106,14 @@ func (me *Engine) Close() {
 func (me *Engine) Subscribe(ctx context.Context, name string, send func(*sqltypes.Result) error) (done <-chan struct{}, err error) {
 	me.mu.Lock()
 	defer me.mu.Unlock()
+	// Return a closed channel if engine is not open.
```

Comment:
I think I'd rather return 'unavailable' here. Then on the vtgate side we can catch that error code. It would be more consistent overall, and easier to differentiate why vttablet didn't respond.

Reply:
Comment is wrong, we don't return a closed channel, but an error.